compass/landice/tests/ismip6_forcing/ocean_thermal/process_thermal_forcing.py

```diff
@@ -0,0 +1,319 @@
+                            target=os.path.join(base_path_ismip6, input_file[0]))
+        self.add_output_file(filename=output_file)
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+
+        section = config['ismip6_ais']
+        mali_mesh_name = section.get('mali_mesh_name')
+        mali_mesh_file = section.get('mali_mesh_file')
+        period_endyear = section.get("period_endyear")
+        model = section.get("model")
+        scenario = section.get("scenario")
+        output_base_path = section.get('output_base_path')
+
+        section = config['ismip6_ais_ocean_thermal']
+        method_remap = section.get('method_remap')
+        process_obs_data = section.getboolean('process_obs_data')
+
+        if process_obs_data:
+            input_file_list = self._file_obs
+            output_file = f'processed_obs_TF_1995-2017_8km_x_60m.nc'
+            output_path = f'{output_base_path}/ocean_thermal_forcing/'\
+                          f'obs'
+        else:
+            input_file_list = self._files[period_endyear][model][scenario]
+            output_file = f'processed_TF_{model}_{scenario}_{period_endyear}.nc'
+            output_path = f'{output_base_path}/ocean_thermal_forcing/' \
+                          f'{model}_{scenario}/1995-{period_endyear}'
+
+        input_file = os.path.basename(input_file_list[0])
+
+        # interpolate and rename the ismip6 thermal forcing data
+        remapped_file_temp = "remapped.nc"  # temporary file name
+
+        # call the function that reads in, remap and rename the file.
+        print("Calling a remapping function...")
+        self.remap_ismip6thermalforcing_to_mali(input_file,
+                                                remapped_file_temp,
+                                                mali_mesh_name,
+                                                mali_mesh_file, method_remap)
+
+        # call the function that renames the ismip6 variables to MALI variables
+        print("Renaming the ismip6 variables to mali variable names...")
+        self.rename_ismip6thermalforcing_to_mali_vars(remapped_file_temp,
+                                                      output_file)
+
+        print("Remapping and renamping process done successfully. "
+              "Removing the temporary file 'remapped.nc'")
+
+        # remove the temporary combined file
+        os.remove(remapped_file_temp)
+
+        # place the output file in appropriate directory
+        if output_base_path == "NotAvailable":
+            return
+
+        if not os.path.exists(output_path):
+            print("Creating a new directory for the output data")
+            os.makedirs(output_path)
+
+        src = os.path.join(os.getcwd(), output_file)
+        dst = os.path.join(output_path, output_file)
+        shutil.copy(src, dst)
+
+    def remap_ismip6thermalforcing_to_mali(self, input_file,
```

Comment:
This is all lowercase, which I like but I would use underscores to separate words:
```suggestion
    def remap_ismip6_thermal_forcing_to_mali(self, input_file,
```